api/txspool.go

```diff
@@ -279,5 +279,13 @@ func (a *API) verifyPoolL2TxWrite(txw *l2db.PoolL2TxWrite) error {
 			))
 		}
 	}
+	// Extra sanity checks: those checks are valid as per the protocol, but are very likely to
+	// have unexpected side effects that could have a negative impact on users
+	switch poolTx.Type {
```

Comment:
sorry, I think I was not clear. You can use the same switch above for this statement

```go
		switch poolTx.Type {
	case common.TxTypeTransfer:
		// ToIdx exists and match token
		toAccount, err := a.h.GetCommonAccountAPI(poolTx.ToIdx)
		if err != nil {
			return tracerr.Wrap(fmt.Errorf("Error getting to account: %w", err))
		}
		if poolTx.TokenID != toAccount.TokenID {
			return tracerr.Wrap(fmt.Errorf("tx.TokenID (%v) != toAccount.TokenID (%v)",
				poolTx.TokenID, toAccount.TokenID))
		}
	case common.TxTypeTransferToEthAddr:
		// ToEthAddr has account created with matching token ID or authorization
		ok, err := a.h.CanSendToEthAddr(poolTx.ToEthAddr, poolTx.TokenID)
		if err != nil {
			return err
		}
		if !ok {
			return tracerr.Wrap(fmt.Errorf(
				"Destination eth addr (%v) has not a valid account created nor authorization",
				poolTx.ToEthAddr,
			))
		}
	case common.TxTypeExit:
		// Extra sanity checks: those checks are valid as per the protocol, but are very likely to
		// have unexpected side effects that could have a negative impact on users
		if poolTx.Amount.Cmp(big.NewInt(0)) <= 0 {
			return tracerr.New(ErrExitAmount0)
		}
	}
	return nil
```